### PR TITLE
Overhaul and fix the weather app

### PR DESCRIPTION
**Pull request by nicomiguelino**
### What this PR does

- [x] Removes the use of `IndexedDB` in favor of `localStorage`.
- [x] Rewrites the app code to use [Alpine.js](https://alpinejs.dev/).
- [x] Makes use of `localStorage` instead
- [x] Provides sensible default values
- [x] Modifies the documentation for installing and deploying this Edge App (if necessary).
- [x] Adds a loading text and icon.

### Screenshots

DIsplay when the player is offline and when there's no data in `localStorage`:

image

### CODE REVIEW

**Comment by nicomiguelino** [comment]
@vpetersson, @MuratUrsavas
Changes are now ready for review again.

I added support for displaying an error message on-screen (not just on the console) if the player's offline and if there's no data to get from the cache. See the screenshot below for reference:

image

Here's some updates as well...
- The weather app still shows the time in 12-hour format for countries that should be using the 24-hour format.
- If I'll use the JS library that I used on the clock app, it would be the same story (I'll not be able to upload the app successfully).

I'd like to hear back from you. Thanks...

**Comment by MuratUrsavas** [comment]
@nicomiguelino What do we do when there is no connection? Do we retry? If not, we certainly should. Relying solely on playlist duration could be problematic as users could choose long periods.

While waiting for that, a simple loading icon and a "Waiting for Internet connection..." message can be shown.

**Comment by MuratUrsavas** [comment]
Approved as my request is optional.

**Review by MuratUrsavas — APPROVED** [review]

**Comment by nicomiguelino** [comment]
@MuratUrsavas, usera can tweak the cache_interval setting, which refers to the interval between data fetch.

Haven'd included a loading indicator (whether it be text or a spinner, for instance). That's a good idea.

**Review by korvyashka — APPROVED** [review]

File: edge-apps/weather/static/js/main.js [hunk 1]
```diff
@@ -1,351 +1,315 @@
...
-    this.openRequest = window.indexedDB.open(this.dbName, 1)
+  constructor({ storeName }) {
+    this.storeName = storeName;
+
+    if (localStorage.getItem(this.storeName) === null) {
+      this.data = {};
+      localStorage.setItem(this.storeName, JSON.stringify(this.data));
+    } else {
+      this.data = JSON.parse(localStorage.getItem(this.storeName));
+      console.log('Database setup complete');
```
[the remark this answers is not in the document]
Comment (nicomiguelino) [reply edge-apps/weather/static/js/main.js:12]: Got it.

File: edge-apps/weather/static/js/main.js [hunk 2]
```diff
@@ -1,351 +1,315 @@
 /* global icons, moment */
 
 class AppCache {
-  constructor({ dbName, storeName }) {
-    this.db = null
-    this.dbName = dbName
-    this.storeName = storeName
-
-    this.openRequest = window.indexedDB.open(this.dbName, 1)
+  constructor({ storeName }) {
```
[the remark this answers is not in the document]
Comment (nicomiguelino) [reply edge-apps/weather/static/js/main.js:4]: Yes, it's just a key in the storage. Thanks for noticing this.
The class-based wrapper was originally for `IndexedDB`.
I'll just rename it to something like `keyName` in future commits.

**Comment by nicomiguelino** [comment]
@MuratUrsavas, @korvyashka, @vpetersson
Please review...

### Screenshots

I also included them in the description section.

image

**Comment by MuratUrsavas** [comment]
Looking good! :+1:

**Comment by MuratUrsavas** [comment]
Although "an" doesn't sound right to me. But I'm not an English expert, obviously :)

**Comment by nicomiguelino** [comment]
https://github.com/Screenly/Playground/assets/10234135/bea70c0a-08c4-467e-b610-357decdad3fb

https://github.com/Screenly/Playground/assets/10234135/33035566-b2dc-45db-a2de-008b6ddd55bb

**Comment by vpetersson** [comment]
@sergey-borovkov what can we do on the edge app side to ensure preloading works as expected?

**Comment by nicomiguelino** [comment]
@MuratUrsavas, @sergey-borovkov, @vpetersson
Would like to follow up on this one... Thanks.

**Pull request merged**